File: frontend/src/app/models/medico.model.ts
import { Usuario } from './usuario.model';

export class Medico extends Usuario {
  constructor(
    id: number,
    nombre: string,
    apellidos: string,
    usuario: string,
    clave: string,
    public numColegiado: string,
    public pacienteIds: number[] = [],
  ) {
    super(id, nombre, apellidos, usuario, clave);
  }
}

File: backend/CitasMedicas.Application/Models/MedicoModel.cs
namespace CitasMedicas.Application.Models;

public class MedicoModel
{
    public int Id { get; set; }

    public string Nombre { get; set; } = string.Empty;

    public string Apellidos { get; set; } = string.Empty;

    public string Usuario { get; set; } = string.Empty;

    public string? Clave { get; set; }

    public string NumColegiado { get; set; } = string.Empty;

    public List<int> PacienteIds { get; set; } = [];
}